### PR TITLE
My Site Dashboard: change site

### PR DESCRIPTION
**Pull request by leandroalonso**
Part of #17873

Changes the content of the dashboard when the selected site changes.

https://user-images.githubusercontent.com/7040243/153418112-3a0017e2-ac7b-4aa2-9ab0-494870d0556f.mp4

### To test

**Make sure to sandbox the API.**

#### Changing sites

1. Open the app and go to Dashboard
2. Check that the content is displayed
3. Change the current site
4. Check that the dashboard update

#### Changing between menu and dashboard

1. Open the app and go to Dashboard
2. Tap Site Menu
3. Tap the dashboard again
4. It should refresh and show the content

#### Checking allocation

1. Open Instruments > Allocations
2. Select the current WordPress running on your simulator
3. In the filter add `PostsCardViewController`
4. Play with the app for some time
5. Check the number of persistent `PostsCardViewController` and compare to the number of persistent `DashboardPostsCardCell`, they should be equal

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by momo-ozawa — COMMENTED** [review]
The first time I change the site on the dashboard, the dashboard gets updated. But if I try switching to another site again, the dashboard doesn't get updated.

File: WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift [hunk 1]
```diff
@@ -575,7 +575,7 @@ class MySiteViewController: UIViewController, NoResultsViewHost {
             blogDetailsViewController?.tableView.reloadData()
             blogDetailsViewController?.preloadMetadata()
         case .dashboard:
-            // TODO: Update blog dashboard vc
+            blogDashboardViewController?.update(blog: blog)
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift:589]: [nit] We can delete the break statement below

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewController.swift [hunk 2]
```diff
@@ -60,6 +66,13 @@ final class BlogDashboardViewController: UIViewController {
         activityIndicatorView.stopAnimating()
     }
 
+    func update(blog: Blog) {
+        self.blog = blog
+        viewModel.blog = blog
+        viewModel.applySnapshotForInitialData()
+        viewModel.start()
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewController.swift]: What do you think of renaming these viewModel methods?
```suggestion
        viewModel.clearSnapshot()
        viewModel.loadCards()
```

**Review by momo-ozawa — APPROVED** [review]
Works as described. Left some minor comments.

File: WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift (hunk 1 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift:589]: Done in 1276a968baa925b7c7306c05fa947c50d31d9fa0

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewController.swift (hunk 2 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewController.swift]: Done in 3cb7f33e994166bca58ebc151382f4baf54fe34c

FYI, `applySnapshotForInitialData` is going to change in the next PR (due to the persistence). I'll update the name there.

**Pull request merged**